src/types.ts

```diff
@@ -957,6 +957,21 @@ export class ZodString extends ZodType<string, ZodStringDef> {
     });
   }
 
+  json<T extends ZodTypeAny>(shape: T) {
+    return this.transform((val, ctx) => {
+      try {
+        return JSON.parse(val);
+      } catch (error: unknown) {
+        ctx.addIssue({
+          code: ZodIssueCode.invalid_string,
+          validation: "json",
+          message: (error as Error).message,
```

Comment:
With a quick scan, can't find anywhere else where `error as Error` is used.
So if you need it to be a `message: string`, check for `instanceof` first, or use the `...errorUtil.errToObj`

Reply:
IMO `as` is better here than doing something at runtime. `JSON.parse(...)` only throws `Error`s (specifically `SyntaxError`s) as [documented by MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/parse#exceptions). I also don't feel particularly strongly though.